eng/Packages.Data.props

```diff
@@ -203,7 +203,7 @@
     All should have PrivateAssets="All" set so they don't become package dependencies
   -->
   <ItemGroup>
-    <PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version="3.0.0-beta.20240229.1" PrivateAssets="All" />
+    <PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version="3.0.0-beta.20240321.4" PrivateAssets="All" />
```

Comment:
The emitter package json should stay in sync here, not sure what if any implications of only updating one of them is